source/multi_source.go

```diff
@@ -23,22 +23,42 @@ type multiSource struct {
 	children []Source
 }
 
+// struct to allow for quickly creating distinct set of endpoints to return
+type endpointKey struct {
+	DNSName string
+	TxtType bool
+}
+
 // Endpoints collects endpoints of all nested Sources and returns them in a single slice.
 func (ms *multiSource) Endpoints() ([]*endpoint.Endpoint, error) {
-	result := []*endpoint.Endpoint{}
+	uniqueEndpoints := make(map[endpointKey]*endpoint.Endpoint)
 
 	for _, s := range ms.children {
 		endpoints, err := s.Endpoints()
 		if err != nil {
 			return nil, err
 		}
 
-		result = append(result, endpoints...)
+		for _, endpoint := range endpoints {
+			uniqueEndpoints[newEndpointKey(endpoint)] = endpoint
+		}
+	}
+
+	result := []*endpoint.Endpoint{}
+	for _, e := range uniqueEndpoints {
+		result = append(result, e)
 	}
 
 	return result, nil
 }
 
+func newEndpointKey(endpoint *endpoint.Endpoint) endpointKey {
+	return endpointKey{
+		DNSName: endpoint.DNSName,
+		TxtType: endpoint.RecordType == "TXT",
```

Comment:
I was trying to support a situation where we need to support AN `A` or `CNAME` record and the associated `TXT` record for the registry. 

I'll take a closer look at the lifecycle and see what additional context the `plan` stage has and if that can be used to make this cleaner.